static/src/javascripts/projects/common/modules/commercial/build-page-targeting.js

```diff
@@ -292,23 +290,28 @@ const buildPageTargetting = (
 
 const getPageTargeting = (): { [key: string]: mixed } => {
     if (Object.keys(myPageTargetting).length !== 0) return myPageTargetting;
+    const onCMPConsentNotification = isInTcfv2Test()
+        ? onConsentChange
+        : oldCmp.onIabConsentNotification;
 
     onCMPConsentNotification(state => {
-        let canRun: boolean;
+        let canRun: boolean | null;
         if (typeof state === 'boolean') {
             // CCPA mode
             canRun = !state;
         } else if (typeof state.tcfv2 !== 'undefined') {
             // TCFv2 mode,
-            canRun = Object.values(state.tcfv2).every(Boolean);
+            canRun = Object.values(state.tcfv2.tcfData).every(p => p === null) ? null :
+                Object.values(state.tcfv2.tcfData).every(Boolean);
```

Comment:
Where does this 'Boolean' function come from?

Reply:
@jeteve : it's in [the spec](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Boolean).

> The value passed as the first parameter is converted to a boolean value, if necessary. If the value is omitted or is 0, -0, null, false, NaN, undefined, or the empty string (""), the object has an initial value of false. All other values, including any object, an empty array ([]), or the string "false", create an object with an initial value of true.

Reply:
@mxdvl beat me to it 🙂

Reply:
So it acts as an identity function for boolean values in this case where the values are themselves boolean?

So it's the same as writing  `.every( (bool) => bool )`

Reply:
This file has now been reverted to not include the `consent_tcfv2`k-v. We will do that on a separated PR which will be easier to review as discussed